### ISSUE TITLE
Script menu does not recognize '#!/usr/bin/swift' filename.swift Unix scripts

### ISSUE THREAD

**Issue by marc-medley** [coteditor/CotEditor#692]
## Environment

- CotEditor: 3.1.7 (188)
- System: macOS Version 10.12.3 (Build 16D32)
- Language: English
- Swift: 3.0.2
## Short Description

CotEditor script menu does not recognize '#!/usr/bin/swift' filename.swift Unix scripts.  And, the Swift file default to a "Shell Script" Syntax Style instead of "Swift" if the extension is `*.sh`.

## Steps to Reproduce the Issue

1. Start with a '#!/usr/bin/swift' Unix script.  s in are available to use in [GitHub 004.45macOS_CotEditorSwiftScripting Repository](https://github.com/marc-medley/004.45macOS_CotEditorSwiftScripting).

or create test files "hello shell extension.sh" and "hello swift extension.swift " with the same content:

``` swift
#!/usr/bin/swift
// %%%{CotEditorXInput=None}%%%
// %%%{CotEditorXOutput=InsertAfterSelection}%%%

import Foundation
print("Hello, Swiff !# Unix script!")
```

Set both files to be executable.

2. Place the executable '#!/usr/bin/swift' "hello shell extension.sh" and "hello swift extension.swift " files in the CotEditor 'Open Scripts Folder' for

3. Notice that .swift file does not show on the CotEditor menu.

4. Close and reopen the `#!/usr/bin/swift` *.sh file.  Notice that the Syntax Style is "Shell Script" instead of "Swift".

## Expected Result

Expected to be able to call `*.swift` files **which contain `#!...swift` as the first line** from the CotEditor scripts menu.

The ability for the Scripts menu to recognize `#!...swift` `*.swift` files will also solve the Syntax Style issue.  

Also note the Swift may be installed in some path other than `/usr/bin`

**Comment by 1024jp** [comment]
Because Swift is currently not supported in the CotEditor scripting. But I can add it. I'll add it in the next CotEditor 3.2.0.

**Comment by marc-medley** [comment]
thank you.

**Comment by 1024jp** [comment]
I changed my mind and shipped this change as a part of CotEditor 3.1.8.
